cookbooks/cdo-varnish/libraries/http_cache.rb

```diff
@@ -0,0 +1,100 @@
+# HTTP Cache configuration.
+#
+# See cdo-varnish/README.md for more information on the configuration format.
+class HttpCache
+  STATIC_ASSETS = {
+    # For static-asset extensions, don't forward any cookies or headers.
+    path: %w(cur pdf png gif jpeg jpg ico mp3 swf css js).map{|ext| "*.#{ext}"},
+    headers: [],
+    cookies: 'none'
+  }
+
+# Host header is needed for Varnish to delegate to pegasus/dashboard depending on the host.
+# Accept-Language header is needed to cache language-specific pages.
+  ALL_HEADERS = %w(Host Accept-Language)
```

Comment:
After looking at the CloudFront [documentation](http://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/RequestAndResponseBehaviorCustomOrigin.html#request-custom-headers-behavior) further and running some tests, I can clarify that any headers NOT listed in that table are considered "Customer-defined headers", which are forwarded to the origin even when a custom header is not included in the cache key.

So CloudFront only removes/changes the headers it explicitly mentions in the table (such as `Accept`, `Accept-Language`, and `User-Agent`), and leaves the rest untouched.